services/one-stream/package/deletePackage.js

```diff
@@ -0,0 +1,64 @@
+import AWS from "aws-sdk";
+
+import { dynamoConfig } from "cmscommonlib";
+
+const dynamoDb = new AWS.DynamoDB.DocumentClient(dynamoConfig);
+
+const oneMacTableName = process.env.IS_OFFLINE
+  ? process.env.localTableName
+  : process.env.oneMacTableName;
+
+export const deletePackage = async (packageId) => {
+  console.log("deleting package: ", packageId);
+  const queryParams = {
+    TableName: oneMacTableName,
+    KeyConditionExpression: "pk = :pk",
+    ExpressionAttributeValues: {
+      ":pk": packageId,
+    },
+  };
+  console.log("%s the new query params are: ", packageId, queryParams);
+
+  try {
+    const result = await dynamoDb.query(queryParams).promise();
+    console.log("%s query result: ", packageId, result);
+    if (result?.Items.length <= 0) {
+      console.log("%s did not have Items?", packageId);
+      return;
+    }
+    await Promise.all(
+      result.Items.map(async (item) => {
+        if (item.GSI1pk) delete item.GSI1pk;
+        if (item.GSI1sk) delete item.GSI1sk;
+        if (item.GSI2pk) delete item.GSI2pk;
+        if (item.GSI2sk) delete item.GSI2sk;
```

Comment:
doesnt make any difference but just noted that we dont need to test for property existence before invoking delete, if the property doesnt exist the delete operation would just ignore and continue

Reply:
Huh, I could've sworn I got exceptions if the attribute didn't exist, but I do sometimes try to predict where an exception could happen, too.... I'll have to try it sometime :)